mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -502,19 +502,19 @@ class Word2VecModel private[spark] (
   private val vectorSize = wordVectors.length / numWords
 
   // wordList: Ordered list of words obtained from wordIndex.
-  private val wordList: Array[String] = {
-    val (wl, _) = wordIndex.toSeq.sortBy(_._2).unzip
-    wl.toArray
+  private lazy val wordList: Array[String] = {
+    wordIndex.toSeq.sortBy(_._2).iterator.map(_._1).toArray
   }
 
   // wordVecNorms: Array of length numWords, each value being the Euclidean norm
   //               of the wordVector.
-  private val wordVecNorms: Array[Float] = {
-    val wordVecNorms = new Array[Float](numWords)
+  private lazy val wordVecNorms: Array[Float] = {
```

Comment:
How much does this save, if it only happens once and has to happen to use the model?

Reply:
this var `wordVecNorms` is only used in method `findSynonyms` in the .mllib.w2v; however, this `findSynonyms`  is never used in the .ml side. So I think we can make it lazy.

Reply:
OK fair enough. There are use cases here that would never need this calculated?

Reply:
`however, this findSynonyms is never used in the .ml side. So I think we can make it lazy.`

I am wrong. this var `wordVecNorms` is used in methods `findSynonyms` and `findSynonymsArray` in the .ml side. Since it is not used in `transform`, so we can still mark it lazy